### ISSUE TITLE
Measure frequency(period) of input step pulses

### ISSUE THREAD

**Issue by IhorNehrutsa** [CAP1Sup/Intellistep#95]
Is it possible to measure frequency(period) of input step pulses on fly by hardware counters?
This allow to continue correction timer with proper(last input step) frequency(period).

**Comment by CAP1Sup** [comment]
I don't think so. I did a large amount of research into the timers and I don't remember ever coming across a period counter. We could do something with software though, it's not too hard, especially considering that everything is perfectly correct with the counter

**Comment by xerootg** [comment]
Yeah, you should be able to measure durations in a timer. It's gonna be
dependent on what pin you want to count on though... Tim2 can definitely do
it, but that's currently counting pulses. It's not a huge jump to say that
frequency is pulses over time, so maybe that's a starting point? Do you
have a copy of the full manual for the stm32f1xx?
